src/ET_Client.php

```diff
@@ -190,14 +197,43 @@ function __construct($getWSDL = false, $debug = false, $params = null)
             {
                 $this->scope = $params['scope'];
             }
+			if (array_key_exists('applicationType', $params))
+			{
+				$this->applicationType = $params['applicationType'];
+			}
+			if (array_key_exists('redirectURI', $params))
+			{
+				$this->redirectURI = $params['redirectURI'];
+			}
+			if (array_key_exists('authorizationCode', $params))
+			{
+				$this->authorizationCode = $params['authorizationCode'];
+			}
 		}
 
 		$this->debugSOAP = $debug;
-		
-		if (!property_exists($this,'clientId') || is_null($this->clientId) || !property_exists($this,'clientSecret') || is_null($this->clientSecret)){
-			throw new Exception('clientid or clientsecret is null: Must be provided in config file or passed when instantiating ET_Client');
+
+		if (!property_exists($this,'applicationType') || is_null($this->applicationType) || empty($this->applicationType)){
+			$this->applicationType = 'server';
 		}
-		
+
+		if($this->applicationType == 'public' || $this->applicationType == 'web'){
+			if (!property_exists($this,'redirectURI') || is_null($this->redirectURI ) || !property_exists($this,'authorizationCode') || is_null($this->authorizationCode)){
```

Comment:
same as above, property_exists does not make sense. also here, you should use empty() instead in case the user just puts an empty string